### PR TITLE
docs: Updates to make the docs easier to digest.

### PR DESCRIPTION
**Pull request by vlidholt**
Cleans up the documentation by:

- Adding videos where relevant.
- Removing content that are out-of-date or not needed.
- Collapsing the reference section, to make docs look more digestable.
- Improving the _Get started_ page and adds a _Get started with Mini_ page.
- More logically grouping some of the pages and adding them sub-sections.

More updates to come, but this feels like a good start. :)

### CODE REVIEW

File: docs/01-get-started.md [hunk 1]
```diff
@@ -1,16 +1,40 @@
...
+<details>
+<summary>__Serverpod vs Serverpod Mini comparison__</summary>
+<p>
+
+| Feature               | Serverpod | Serverpod Mini |
+|-----------------------|-----------|:--------------:|
+| Remote method calls   |     ✅     |        ✅       |
+| Generated data models |     ✅     |        ✅       |
+| Streaming data        |     ✅     |        ✅       |
+| Social authentication |     ✅     |                |
```
Comment (SandPod) [inline docs/01-get-started.md:22]: Question: What does this mean? Serverpod Mini still supports authentication; it just doesn't support our authentication module.

You could, for example, build a JWT token validation authentication without having to store anything in the database.

This make it sound like social authentication is not possible with mini.

File: docs/02-get-started-with-mini.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+For types, you can use most basic Dart types, such as `String`, `double`, `int`, `bool`, `DateTime`, and `ByteData`. You can also include `List` and `Map`, just make sure to specify their types. Any other class specified among your models is also supported.
+
+Whenever you add or edit a model file, run `serverpod generate` in your server directory. Then, Serverpod will generate all the updated Dart classes:
+
+```bash
+$ cd myminipod/myminipod_server
+$ serverpod generate
+```
+
+## Adding methods to your server
```
Comment (SandPod) [inline docs/02-get-started-with-mini.md:55]: Comment: The story based instructions are great! We should make this improvement to the non mini "Get Started" page as well.

File: docs/02-get-started-with-mini.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+To call the endpoint method we just created from Flutter, just create a `Company` object, call the method, and await the result:
+
+```dart
+var company = Company(
+  name: 'Serverpod',
+  foundedDate: DateTime(2021, 9, 27),
+  employees: [
+    'Alex',
+    'Isak',
+    'Viktor',
```
Comment (SandPod) [inline docs/02-get-started-with-mini.md:103]: 🥳

File: docs/07-cli.md [hunk 4]
```diff
@@ -1,31 +0,0 @@
-# Serverpod CLI
```
Comment (SandPod) [inline docs/07-cli.md]: Question: This seems to have been removed. Is the thought that the CLI should have all documentation through its help method?

I see a use-case for some additional information on the CLI commands that might need to be explained in more detail.

File: versioned_docs/version-1.1.0/index.md [hunk 5]
```diff
@@ -34,7 +34,7 @@ Serverpod Insights is a companion app bundled with Serverpod. It allows you to a
 
 :::info
 
-Download the latest version here: __[Serverpod Insights](insights)__. It is compatible with Serverpod version 1.0.x. Always use the same version of Serverpod Insights as for the framework itself.
+Download the latest version here: __[Serverpod Insights](tools/insights)__. It is compatible with Serverpod version 1.0.x. Always use the same version of Serverpod Insights as for the framework itself.
```
Comment (SandPod) [inline versioned_docs/version-1.1.0/index.md:37]: Question: When testing this the reference now jumps to the newest version of the documentation if the link is used. Meaning that the visitor will leave the 1.x.x version of the docs and jump to 2.0.0 (currently). Was that intentional?

Comment (SandPod) [reply versioned_docs/version-1.1.0/index.md:37]: Same for 1.1.1 and 1.2.0.

**Review by SandPod — COMMENTED** [review]
Awesome improvement! Looking forward to the next of these!

Just a couple of questions.

Also, since you are shuffling some files around, have you validated that all links contiune to work? The way I did this before was to compare the previosly generated sitemap with the newest one.

File: docs/01-get-started.md (hunk 1 above)
Comment (vlidholt) [reply docs/01-get-started.md]: Clarified.

File: versioned_docs/version-1.1.0/index.md (hunk 5 above)
Comment (vlidholt) [reply versioned_docs/version-1.1.0/index.md:37]: Can't seem to fix this. But may not be a big issue.

**Review by SandPod — APPROVED** [review]
LGTM! 👍

**Pull request merged**